sdk/iot_hub/src/service/operations/create_or_update_device_identity.rs

```diff
@@ -67,17 +66,23 @@ impl CreateOrUpdateDeviceIdentityBuilder {
             let body = azure_core::to_json(&body)?;
             request.set_body(body);
 
-            self.client
-                .http_client()
-                .execute_request_check_status(&request)
-                .await?
-                .try_into()
+            let response = self.client.send(&mut self.context, &mut request).await?;
+
+            CreateOrUpdateDeviceIdentityResponse::try_from(response).await
         })
     }
 }
 
 pub type CreateOrUpdateDeviceIdentityResponse = DeviceIdentityResponse;
 
+impl CreateOrUpdateDeviceIdentityResponse {
+    pub(crate) async fn try_from(response: azure_core::Response) -> azure_core::Result<Self> {
```

Comment:
I think this should probably live with the definition of `DeviceIdentityResponse`

Reply:
done.